GrandCentralBoard/Widgets/Watch/GoogleCalendarWatchWidgetBuilder.swift

```diff
@@ -14,15 +14,23 @@ final class GoogleCalendarWatchWidgetBuilder : WidgetBuilding {
     let name = "googleCalendarWatch"
 
     func build(settings: AnyObject) throws -> Widget {
+        let timeSettings = try TimeSourceSettings.decode(settings)
+        let calendarSettings = try GoogleCalendarSourceSettings.decode(settings)
+
         let networkRequestManager = Manager()
+        let tokenProvider = GoogleTokenProvider(clientID: calendarSettings.clientID,
+                                                clientSecret: calendarSettings.clientSecret,
+                                                refreshToken: calendarSettings.refreshToken)
+        let apiDataProvider = GoogleAPIDataProvider(tokenProvider: tokenProvider, networkRequestManager: networkRequestManager)
+        let calendarDataProvider = GoogleCalendarDataProvider(dataProvider: apiDataProvider)
```

Comment:
tokenProvider - provides new accessToken from refreshToken
apiDataProvider - general provider of data from any wanted endpoint. Makes sure access token is still valid before sending request
calendarDataProvider - uses apiDataProvider to fetch data from calendar related endpoints